### ISSUE TITLE
rest api server running fail

### ISSUE THREAD

**Issue by wormcc** [emqtt/emqttd#1661]
#### Environment

- OS: centos6.5
- Erlang/OTP: 
- EMQ: 2.3

#### Description
the listener for the rest api not running, but when i use the command `emqttd_ctl listeners start mqtt:api 8099`, it showed `Failed to restart mqtt:api listener on 8099, error:not_found ` , 8099 is the port for the api server in the config file.

How should i start the server? Thx

**Comment by terry-xiaoyu** [comment]
The mqtt:api port defaults to `8080`. You can't start the api port because someone else has already listened on it, right ?

You could change the configuration and restart the emq for now, I'll look at the `emqttd_ctl listeners restart` problem.

```
## Default: 0.0.0.0:8080
listener.api.mgmt = 0.0.0.0:8099
```

**Comment by Gilbert-Wong** [comment]
I don't know what you want. The listener for rest api would be started after emqttd starting unless the 8080 port has been occupied by other application. If you want to to change the rest api port, change it in  `etc/emq.conf`, then restart emqttd. 

If you want to change listen port of rest api when emqttd is running, it is a new feature for us

**Comment by Gilbert-Wong** [comment]
@terry-xiaoyu  I don't think this is bug. It looks like @wormcc 's problem. Maybe he needs a new feature. But I assume that this is not a good idea which want to change rest api listener's port when emqttd has been started.

**Comment by wormcc** [comment]
@Gilbert-Wong Thanks for the reply, i have changed the default port 8080 to 8099 in the emq.conf, then i restarted the broker, but the rest api server did not run, and the 8099 port  wasn't occupied by other application either.

**Comment by Gilbert-Wong** [comment]
Execute `./bin/emqttd_ctl listeners` and tell me what's printed on terminal
